Genomic pan-cancer classification using image-based deep learning
=====

Setup
------
To run this code you need the following:
- a machine with multiple GPUs (CUDA 9.0)
```
conda create -n tensorflow python=3.6
pip install tensorflow-gpu==1.12.0 numpy==1.16.4 keras==2.2.5 opencv-python scikit-learn matplotlib
```

Reproduce the results of classification experiments
------

```Python
cd DNN-models
# download 'h5 folder'
python 5_load_model.py
```
***The h5 folder of the pre-trained four models see: https://drive.google.com/drive/folders/1Mj0x1Tyhp6jEWVTh2L9vOHtgBnox_T2E?usp=sharing***

Retraining the models of classification
------

```Python
cd DNN-models
# The default model used is Inception-ResNet-v2, change the adopted model by modifying the variable base_model="?"
python 4_train_model.py 
```

Code Details
------

#### Generate-all-mutation-map
###### including original dataset and the code of genetic mutation map construction

```Python
cd Generate-all-mutation-map
python 0_statistic_cancer_geneNums.py
python 1_statistic_variant_gene.py
python 2_statistic_same_gene_chrom_noselect.py
python 3_data_preprocess_noselect_white.py
```

##### Notes:
1. **0_statistic_cancer_geneNums.py:** Statistics the number of genes and save the gene name of each cancer to the directory "/geneNameCancer/" 
2. **1_statistic_variant_gene.py:** Statistics the gene name list of each chromosome for each cancer separately 
	and save it to the corresponding cancer directory under "ChromosomeGene/", 
	such as "ChromosomeGene/ACC/Chrom01.txt"
3. **2_statistic_same_gene_chrom_noselect.py:** Statistics the gene name list of 36 types of cancer under the same chromosome 
	and save it to the corresponding chromosome file under "ChromosomeGene/All/", 
	such as "ChromosomeGene/All/chrom01.txt"
4. **3_data_preprocess_noselect_white.py:** Read the 24 chromosome files under "ChromosomeGene/All/", 
	build a gene dictionary for each chromosome file, 
	calculate the number of columns required for the gene list of each chromosome, 
	and splice into a 310 * 310 matrix in the order of chromosomes. 
	Read the sample file and generate 310 * 310 mutation map for each sample for each cancer under the "dataset/" path.

#### DNN-models
###### including the code of deep neural networks and Guided Grad-CAM visualization

```Python
# Split the dataset ("dataset/" path generated above) into training set, validation set and testing set manually according to the appropriate ratio (eg. 8:1:1)

cd DNN-models
cp -r ../Generate-all-mutation-map/ChromosomeGene/ .
cp -r ../Generate-all-mutation-map/geneNameCancer/ .
python 4_train_model.py

# copy .h5 file into '../h5' folder.
python 5_load_model.py
python 6_draw_heatmap_guidedGradCAM.py
python 7_generate_and_statistics_heatmapt.py
```

##### Notes:
1. **4_train_model.py:** Use the testing set to evaluate the model and save the h5 file, calculate the F1 value
    Decide what model to use by modifying the variable base_model="?"
2. **5_load_model.py:** Reproduce the experimental results based on the h5 file and calculate the F1 value
    Decide what model to use by modifying the variable base_model="?"
3. **6_draw_heatmap_guidedGradCAM.py:** Generate Guided Grad-CAM heatmap and save it to the "Guided Grad-CAM" folder 
    Decide what model to use by modifying the variable base_model="?"
    Decide generate the heatmap by modifying the variable path="?", eg. path= current_path + '/dataset/test' or  current_path + '/dataset/test/BRCA'
4. **7_generate_and_statistics_heatmapt.py:** Generate a heatmap for each cancer heatmap samples 
			and statistics the genes corresponding to the importance score (pixel value) of the heatmap
			need "/ChromosomeGene/All/" and "geneNameCancer" folder generated by Generate-all-mutation-map project



#### Dataset preprocessing:
  Considering that the pixels (genes) in the sample map directly generated from the mutation genes are still extremely sparse, it is difficult to form meaningful maps and is not conducive to the training of deep learning models. Therefore, we finally superimposed all genes to form a complete mutation map for each cancer type by progressively stacking genes from each patient sample into the map. The genes of other patients which have been converted into the map in the previous stages will not be reset in the current stage. Experimental results show that the proposed progressively stacking method can achieve better results.
